src/python/pants/goal/run_tracker.py

```diff
@@ -371,7 +371,7 @@ def _get_headers(cls, stats_version: int) -> Dict[str, str]:
             }
 
   @classmethod
-  def post_stats(cls, stats_url, stats, timeout=2, auth_provider=None, stats_version=1):
+  def post_stats(cls, stats_url: str, stats: dict, timeout: int=2, auth_provider: Optional[str]=None, stats_version: int=1):
```

Comment:
(Carry over from conversation in Toolchain repo). Please use `Dict` here (it's already imported), as it's the convention in this repo.

Also, I hate that we don't yet have Black to avoid us wasting time on this, but there should be spaces between the type hint and default args:

`timeout: int = 2` vs. `timeout: int=2`

Reply:
I don't think that using Dict (instead of dict) make a ton of sense, since the underlaying values in the dict can be sub-dicts or other data types that are then handled by `RunTrackerOptionEncoder`
I think the best we can do here is `Dict[str, Any]` which is not much better than `dict`.

Reply:
It is the convention in Pants to use the `typing` types for collection type annotations, not the type constructors. For uniformity's sake, let's stick to that convention in the future.

(Also, in this case `Dict[str, Any]` would be better than `dict`, as it at least types the keys, but regardless, the main concern here is uniformity).